pkg/genericapiserver/storage_factory.go

```diff
@@ -29,11 +29,21 @@ import (
 	"github.com/golang/glog"
 )
 
-// StorageDestinations takes a GroupResource and returns back its storage interface.  This result includes:
+// StorageFactory is the interface to locate the storage for a given GroupResource
+type StorageFactory interface {
+	// New finds the storage destination for the given group and resource. It will
+	// return an error if the group has no storage destination configured.
+	New(groupResource unversioned.GroupResource) (storage.Interface, error)
+	// Backends gets all backends for all registered storage destinations.
+	// Used for getting all instances for health validations.
+	Backends() []string
+}
+
+// DefaultStorageFactory takes a GroupResource and returns back its storage interface.  This result includes:
 // 1. Merged etcd config, including: auth, server locations, prefixes
 // 2. Resource encodings for storage: group,version,kind to store as
 // 3. Cohabitating default: some resources like hpa are exposed through multiple APIs.  They must agree on 1 and 2
-type StorageDestinations struct {
+type DefaultStorageFactory struct {
```

Comment:
Add `var _ StorageFactory := &DefaultStorageFactory{}` to ensure that DefaultStorageFactory implements StorageFactory interface.